project_task_pull_request_state/models/project_task.py

```diff
@@ -0,0 +1,41 @@
+from odoo import fields, models, api
+
+
+class ProjectTaskState(models.Model):
+    _inherit = "project.task"
+
+    pr_state = fields.Selection(
+        selection=lambda self: self._selection_pr_state(),
+        string="PR State",
+        )
+
+    @api.model
+    def _selection_pr_state(self):
+        """Function to select the state of the pull request"""
+        return [
+            ('open', 'Open'),
+            ('draft', 'Draft'),
+            ('merged', 'Merged'),
+            ('closed', 'Closed'),
+        ]
+
+    @api.model
+    def create(self, vals):
+        if 'project_id' in vals and 'pr_uri' in vals:
+            search_pr_state = self.env['project.project'].search(
+                [('id', '=', vals.get('project_id'))]).pr_state_default
+            vals['pr_state'] = search_pr_state
+        else:
+            vals['pr_state'] = None
+        return super(ProjectTaskState, self).create(vals)
+
+    def write(self, vals):
+        self._check_pr_state(vals)
+        return super(ProjectTaskState, self).write(vals)
+
+    def _check_pr_state(self, vals):
+        if 'project_id' in vals:
```

Comment:
Why are you checking for "project_id" here? What if only "pr_uri" is updated during write?